torch/_dynamo/compiled_autograd.py

```diff
@@ -183,3 +196,108 @@ def disable():
     prior = torch._C._dynamo.compiled_autograd.set_autograd_compiler(None)
     yield
     torch._C._dynamo.compiled_autograd.set_autograd_compiler(prior)
+
+
+# Invoke backward hook is a higher order op meant for both invoking a bound hook,
+# and for registering it as a call_function in the backward graph.
+# This allows us to re-enter dynamo during compiled autograd to trace (or graph break)
+# the hook as needed. This, in turn, means we can support hooks in backward with complex python
+# state mutation. If we were to not do this, the hooks would get inlined into their composing aten ops,
+# and we would lose the python state mutation.
+def _invoke_in_backward(*args, fn, reenter):
+    return _invoke_in_backward_op(*args, fn=fn, reenter=reenter)
```

Comment:
This is a terrible name. This sounds like "run this function when the backwards is called." But it doesn't do that at all, since this operator doesn't implement autograd!